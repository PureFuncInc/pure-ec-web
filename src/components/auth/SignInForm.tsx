import React from 'react'
import { useForm } from 'react-hook-form'
import { z } from 'zod'
import { zodResolver } from '@hookform/resolvers/zod/dist/zod'
import type { BaseFormProps } from '@/types/BaseFormProps'
import { Input } from '@/components/shared/form/Input'

export type SignInFormData = z.infer<typeof SignInFormSchema>

export type SignInFormProps = BaseFormProps<SignInFormData>

export const SignInForm: React.FC<SignInFormProps> = ({
  onSubmit,
}) => {
  const { register, handleSubmit } = useForm<SignInFormData>({
    resolver: zodResolver(SignInFormSchema),
  })

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <label htmlFor='email'>
        Email
      </label>
      <Input
        type='email'
        variant='primary'
        {...register('email')} />

      <label htmlFor='password'>
        Password
      </label>
      <Input
        type='password'
        {...register('password')} />

      <button type='submit'>
        Submit
      </button>
    </form>
  )
}

const SignInFormSchema = z.object({
  email: z.string()
    .email(),
  password: z.string()
    .min(8),
})
