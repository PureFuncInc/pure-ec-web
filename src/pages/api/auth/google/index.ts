import nextConnect from 'next-connect'
import passport from 'passport'
import { passportMiddleware } from '@/utilities/middlewares/passport'

export default nextConnect()
  .use(passportMiddleware)
  .get(passport.authenticate('google', { scope: [ 'openid profile email' ] }))
