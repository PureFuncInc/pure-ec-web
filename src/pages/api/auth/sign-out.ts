import type { IncomingMessage, ServerResponse } from 'http'
import nextConnect from 'next-connect'
import { ironSession } from 'iron-session/express'
import { ironSessionOptions } from '@/utilities/session'

export default nextConnect()
  .use(ironSession(ironSessionOptions))
  .use((req: IncomingMessage, res: ServerResponse) => {
    req.session.destroy()

    res.statusCode = 200
    res.end()
  })
