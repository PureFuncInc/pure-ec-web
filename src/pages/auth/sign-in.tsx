import type { NextPage } from 'next'
import type { SignInFormData } from '@/components/auth/SignInForm'
import { SignInForm } from '@/components/auth/SignInForm'

const SignInPage: NextPage = () => {
  const handleSubmit = (formData: SignInFormData) => {
    console.debug({ formData })
  }

  return (
    <main>
      <SignInForm onSubmit={handleSubmit} />
    </main>
  )
}

export default SignInPage
