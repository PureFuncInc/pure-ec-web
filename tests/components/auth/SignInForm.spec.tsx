import { render } from '@testing-library/react'
import { SignInForm } from '@/components/auth/SignInForm'

describe('SignInForm', () => {
  it('should render component', () => {
    const { container } = render(<SignInForm onSubmit={jest.fn()} />)

    expect(container)
      .toMatchSnapshot()
  })
})
