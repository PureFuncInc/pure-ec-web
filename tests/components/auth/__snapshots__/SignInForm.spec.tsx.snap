// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm should render component 1`] = `
<div>
  <form>
    <label
      for="email"
    >
      Email
    </label>
    <input
      class="c-cooaiI c-cooaiI-dXoRhu-variant-primary"
      name="email"
      type="email"
    />
    <label
      for="password"
    >
      Password
    </label>
    <input
      class="c-cooaiI"
      name="password"
      type="password"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
